infrap4d/CMakeLists.txt

```diff
@@ -5,24 +5,22 @@ project(infrap4d VERSION 0.1.0 LANGUAGES C CXX)
 set(STRATUM_TDI_BIN_DIR ${STRATUM_SOURCE_DIR}/stratum/hal/bin/tdi)
 set(STRATUM_BFRT_BIN_DIR ${STRATUM_SOURCE_DIR}/stratum/hal/bin/barefoot)
 
-if(DPDK_TARGET)
-    set(INFRAP4D_MAIN ${STRATUM_TDI_BIN_DIR}/dpdk/main.cc)
-elseif(TOFINO_TARGET)
-    set(INFRAP4D_MAIN ${STRATUM_TDI_BIN_DIR}/tofino/main.cc)
-endif()
+add_executable(infrap4d infrap4d_main.cc)
```

Comment:
This structure doesn't allow for the possibility of building without krnlmon.

My recommendation would be to think in terms of the following:
```
if(WITH_KRNLMON)
    add_executable(infrap4d infrap4d_main.cc)
elseif(DPDK_TARGET)
    add_executable(${STRATUM_TDI_BIN_DIR}/dpdk/main.cc)
elseif(TOFINO_TARGET)
    add_executable(${STRATUM_TDI_BIN_DIR}/tofino/main.cc)
endif()
```
Use `infrap4d_main.cc` only when you're building stratum+srnlmon; otherwise, use the native `main.cc` file.

It's more direct, it removes the need to conditionalize `infrap4d_main.cc` for the non-krnlmon case, and it gives you better parity with the Bazel build.